packages/langium/src/grammar/langium-grammar.langium

```diff
@@ -11,7 +11,43 @@ entry Grammar:
         (definesHiddenTokens?='hidden' '(' (hiddenTokens+=[AbstractRule:ID] (',' hiddenTokens+=[AbstractRule:ID])*)? ')')?
     )?
 	imports+=GrammarImport*
-	(rules+=AbstractRule)+;
+	(rules+=AbstractRule | interfaces+=Interface | types+=Type)+;
+
+///////////////////////////////////////////////////////////////////////////////
+
+// interface ///////
+Interface:
+	'interface' name=ID
+	('extends' superTypes+=[AbstractType] (',' superTypes+=[AbstractType])*)?
+	SchemaType;
+
+fragment SchemaType:
+	'{' attributes+=TypeAttribute* '}' ';'?;
+
+TypeAttribute:
+	name=ID (isOptional?='?')? ':' ^type=AtomType ';'?;
+
+// type ////////////
+Type:
+	'type' name=ID '=' typeAlternatives+=AtomType ('|' typeAlternatives+=AtomType)* ';'?;
```

Comment:
Using `AtomType` here is problematic because `Type` is a subtype of `AbstractType`, which is used for cross-reference targets and for supertypes of interfaces, but these may only be object types. We can solve this with validation forbidding the usage of primitive types, references and arrays here.

Reply:
Should be done as part of #415.